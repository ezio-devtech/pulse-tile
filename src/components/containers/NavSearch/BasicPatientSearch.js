import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import qs from 'qs';

import PTButton from '../../ui-elements/PTButton/PTButton';
import { clientUrls } from '../../../config/client-urls.constants';

export default class BasicPatientSearch extends PureComponent {
    state = {
      searchString: '',
    };

    static contextTypes = {
      router: PropTypes.shape({
        history: PropTypes.object,
      }),
    };

    handleSearchStringChange = event => this.setState({ searchString: event.target.value });

    handleClearSearchString = () => this.setState({ searchString: '' });

    handleSearchClick = () => {
      const { searchString } = this.state;
      if (searchString) {
        const queryParams = {
          orderType: 'ASC',
          pageNumber: 1,
          searchString,
          queryType: 'Patient: ',
        };
        const patientsFullDetailsUrl = `${clientUrls.PATIENTS_FULL_DETAILS}?${qs.stringify(queryParams)}`;

        //TODO prevent "Warning: Hash history cannot PUSH the same path; a new entry will not be added to the history stack"
        this.context.router.history.push(patientsFullDetailsUrl);
      }
    };

    render() {
      const { searchString } = this.state;

      return (
        <div>
          <PTButton className="btn btn-search" onClick={this.handleSearchClick}>
            <i className="btn-icon fa fa-search" />
          </PTButton>
          <div className="wrap-search-holder">
            <div className="search-holder">
              <form>
                <input className="form-control" placeholder="Search" type="text" value={searchString} onChange={this.handleSearchStringChange} />
              </form>
              {/*TODO: should be button*/}
              { searchString && <i className="clearAll fa fa-times-circle" onClick={this.handleClearSearchString} />}
            </div>
          </div>
        </div>
      )
    }
}
