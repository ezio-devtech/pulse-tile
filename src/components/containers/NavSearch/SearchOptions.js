import React from 'react';
import PropTypes from 'prop-types';

const SearchContent = ({ onSelect, BASIC_SEARCH, ADVANCED_SEARCH }) =>
  <div className="dropdown-menu dropdown-menu-search-select dropdown-menu-panel dropdown-menu-left dropdown-menu-small-size">
    <div className="heading">Search Options</div>
    <div className="dropdown-menu-wrap-list">
      <div className="dropdown-menu-list">
        <div className="dropdown-menu-item" onClick={() => onSelect(BASIC_SEARCH)} ><span className="dropdown-menu-item-text">Patient Search - Basic</span></div>
        <div className="dropdown-menu-item" onClick={() => onSelect(ADVANCED_SEARCH)}><span className="dropdown-menu-item-text">Patient Search - Advanced</span></div>
        <div className="dropdown-menu-item"><span className="dropdown-menu-item-text">Clinical Query</span></div>
      </div>
    </div>
  </div>

SearchContent.propTypes = {
  onSelect: PropTypes.func.isRequired,
  BASIC_SEARCH: PropTypes.string.isRequired,
  ADVANCED_SEARCH: PropTypes.string.isRequired,
};

export default SearchContent;
