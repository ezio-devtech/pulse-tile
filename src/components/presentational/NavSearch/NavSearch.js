import React from 'react';
import Dropdown from 'simple-react-dropdown'

import SearchContent from '../SearchContent/SearchContent'
import PTButton from '../../ui-elements/PTButton/PTButton';

const NavSearch = () =>
  <div className="wrap-search wrap-header-search">
    <div className="header-search">
      <div className="control-group left control-search-select dropdown">
        <Dropdown content={<SearchContent />}>
          <PTButton className="btn btn-dropdown-toggle btn-search-toggle">
            <i className="btn-icon fa fa-bars" />
          </PTButton>
        </Dropdown>
      </div>
      <PTButton className="btn btn-search">
        <i className="btn-icon fa fa-search" />
      </PTButton>
      <div className="wrap-search-holder">
        <div className="search-holder">
          <form>
            <input className="form-control" placeholder="Search" type="text" />
          </form>
          <i className="clearAll fa fa-times-circle" />
        </div>
      </div>
    </div>
  </div>

export default NavSearch;
