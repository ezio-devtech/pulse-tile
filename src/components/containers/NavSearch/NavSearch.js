import React, { PureComponent } from 'react';
import classNames from 'classnames';

import SearchOptions from './SearchOptions'
import PTButton from '../../ui-elements/PTButton/PTButton';
import BasicPatientSearch from '../BasicPatientSearch/BasicPatientSearch';
import AdvancedPatientSearch from '../AdvancedPatientSearch/AdvancedPatientSearch';
import ClinicalQuerySearch from '../ClinicalQuerySearch/ClinicalQuerySearch';
import PropTypes from 'prop-types';

const BASIC_SEARCH = 'basicSearch';
const ADVANCED_SEARCH = 'advancedSearch';
const SEARCH_CONTENT = 'searchContent';

export default class NavSearch extends PureComponent {
  state = {
    selected: BASIC_SEARCH,
    openedPanel: '',
    openedSearchPanel: true,
  };

  handleSelect = (selected) => {
    this.setState({ selected });
    //TODO remove this spike to close dropdown
    document.body.click();
  };

  componentWillMount() {
    document.addEventListener('click', this.handleClick, false);
    this.changeResolution();
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClick, false);
  }

  handleClick = (e) => {
    if (!this.node.contains(e.target)) {
      this.setState({ openedPanel: '' });
    }
  };

  handleMouseDown = (name) => {
    this.setState((prevState) => {
      if (prevState.openedPanel !== name) {
        return ({ openedPanel: name })
      }
      return ({ openedPanel: '' })
    })
  };

  runSearch = () => {
    this.setState(prevState => ({
      openedSearchPanel: !prevState.openedSearchPanel,
    }));
  };

  changeResolution = () => {
    if (window.innerWidth < 768) {
      this.state.openedSearchPanel = false;
    }
  };


  render() {
    const { selected, openedPanel, openedSearchPanel } = this.state;
    console.log(openedSearchPanel);

    return (
      <div className="header-search-container">
        <NavSearchButton runSearch={this.runSearch} />
        <div className="wrap-search wrap-header-search" ref={node => this.node = node}>
          { openedSearchPanel &&
          <div className="header-search">
            <div
              className={classNames('control-group left control-search-select dropdown', { 'open': openedPanel === SEARCH_CONTENT })}>
              <div>
                <SearchOptions onSelect={this.handleSelect} {...{
                  selected,
                  BASIC_SEARCH,
                  ADVANCED_SEARCH,
                  SEARCH_CONTENT
                }} />
                <PTButton className="btn btn-dropdown-toggle btn-search-toggle"
                          onClick={() => this.handleMouseDown(SEARCH_CONTENT)}>
                  <i className="btn-icon fa fa-bars"/>
                </PTButton>
              </div>
            </div>
            {selected === BASIC_SEARCH && <BasicPatientSearch/>}
            {selected === ADVANCED_SEARCH &&
            <AdvancedPatientSearch className="advanced-search" onClose={() => this.handleSelect(BASIC_SEARCH)}/>}
            {selected === SEARCH_CONTENT &&
            <ClinicalQuerySearch className="clinical-query-search" onClose={() => this.handleSelect(BASIC_SEARCH)}/>}
          </div>
          }
        </div>
      </div>

    )
  }
};

const NavSearchButton = ({ runSearch }) => {
  return (
    <PTButton title="Search" id="search" className="btn-header search-btn" onClick={() => runSearch()}>
      <i className="fa fa-search" />
    </PTButton>
  );
};
NavSearchButton.propTypes = {
  runSearch: PropTypes.func,
};
NavSearchButton.defaultProps = {
  runSearch: function () { },
};
