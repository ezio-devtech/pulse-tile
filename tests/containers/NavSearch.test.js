import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { Provider } from 'react-redux'
import { StaticRouter } from 'react-router'
import configureStore from 'redux-mock-store';

import NavSearch from '../../src/components/containers/NavSearch/NavSearch';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({
  runtime: {},
  userAccount: { role: 'IDCR' },
});
const context = {};
const userAccountIDCR = { role: 'IDCR' };
const userAccountPHR = { role: 'PHR' };

describe('Component <NavSearch />', () => {
  it('should renders correctly', () => {
    const navSearch = mount(
      <Provider store={store}>
        <StaticRouter location="someLocation" context={context}>
          <NavSearch
            userAccount={userAccountIDCR}
          />
        </StaticRouter>
      </Provider>
    );
    navSearch.find('.dropdown-menu-item').at(1).simulate('click');
    navSearch.find('.btn-search-toggle').at(0).simulate('click');
    navSearch.find('.dropdown-menu-item').at(0).simulate('click');
    expect(navSearch).toMatchSnapshot();
  });
  it('should renders correctly', () => {
    const navSearch = mount(
      <Provider store={store}>
        <StaticRouter location="someLocation" context={context}>
          <NavSearch
            userAccount={userAccountPHR}
          />
        </StaticRouter>
      </Provider>
    );
    navSearch.find('.btn-search-toggle').at(0).simulate('click');
    expect(navSearch).toMatchSnapshot();
  });
  it('should renders correctly shallow', () => {
    const navSearch = shallow(
      <NavSearch
        userAccount={userAccountIDCR}
      />
    );
    navSearch.setState({ selected: 'advancedSearch' });
    navSearch.find('.advanced-search').simulate('click');
    expect(navSearch).toMatchSnapshot();
  });
});

