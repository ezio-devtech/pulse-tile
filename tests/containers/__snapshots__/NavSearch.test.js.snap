// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <NavSearch /> should renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <StaticRouter
    basename=""
    context={Object {}}
    location="someLocation"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "someLocation",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NavSearch
        userAccount={
          Object {
            "role": "IDCR",
          }
        }
      >
        <div
          className="wrap-search wrap-header-search"
        >
          <div
            className="header-search"
          >
            <div
              className="control-group left control-search-select dropdown"
            >
              <div>
                <SearchContent
                  ADVANCED_SEARCH="advancedSearch"
                  BASIC_SEARCH="basicSearch"
                  SEARCH_CONTENT="searchContent"
                  onSelect={[Function]}
                  selected="basicSearch"
                >
                  <div
                    className="dropdown-menu dropdown-menu-search-select dropdown-menu-panel dropdown-menu-left dropdown-menu-small-size"
                  >
                    <div
                      className="heading"
                    >
                      Search Options
                    </div>
                    <div
                      className="dropdown-menu-wrap-list"
                    >
                      <div
                        className="dropdown-menu-list"
                      >
                        <div
                          className="dropdown-menu-item active"
                          onClick={[Function]}
                        >
                          <span
                            className="dropdown-menu-item-text"
                          >
                            Patient Search - Basic
                          </span>
                        </div>
                        <div
                          className="dropdown-menu-item"
                          onClick={[Function]}
                        >
                          <span
                            className="dropdown-menu-item-text"
                          >
                            Patient Search - Advanced
                          </span>
                        </div>
                        <div
                          className="dropdown-menu-item"
                        >
                          <span
                            className="dropdown-menu-item-text"
                          >
                            Clinical Query
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </SearchContent>
                <PTButton
                  className="btn btn-dropdown-toggle btn-search-toggle"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-dropdown-toggle btn-search-toggle"
                    onClick={[Function]}
                  >
                    <i
                      className="btn-icon fa fa-bars"
                    />
                  </button>
                </PTButton>
              </div>
            </div>
            <BasicPatientSearch>
              <div>
                <PTButton
                  className="btn btn-search"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-search"
                    onClick={[Function]}
                  >
                    <i
                      className="btn-icon fa fa-search"
                    />
                  </button>
                </PTButton>
                <div
                  className="wrap-search-holder"
                >
                  <div
                    className="search-holder"
                  >
                    <form
                      onSubmit={[Function]}
                    >
                      <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="Search"
                        type="text"
                        value=""
                      />
                    </form>
                  </div>
                </div>
              </div>
            </BasicPatientSearch>
          </div>
        </div>
      </NavSearch>
    </Router>
  </StaticRouter>
</Provider>
`;

exports[`Component <NavSearch /> should renders correctly 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <StaticRouter
    basename=""
    context={Object {}}
    location="someLocation"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "someLocation",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NavSearch
        userAccount={
          Object {
            "role": "PHR",
          }
        }
      >
        <div
          className="wrap-search wrap-header-search"
        >
          <div
            className="header-search"
          >
            <div
              className="control-group left control-search-select dropdown open"
            >
              <PTButton
                className="btn btn-dropdown-toggle btn-search-toggle"
                onClick={[Function]}
              >
                <button
                  className="btn btn-dropdown-toggle btn-search-toggle"
                  onClick={[Function]}
                >
                  <i
                    className="btn-icon fa fa-bars"
                  />
                </button>
              </PTButton>
            </div>
            <BasicPatientSearch>
              <div>
                <PTButton
                  className="btn btn-search"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-search"
                    onClick={[Function]}
                  >
                    <i
                      className="btn-icon fa fa-search"
                    />
                  </button>
                </PTButton>
                <div
                  className="wrap-search-holder"
                >
                  <div
                    className="search-holder"
                  >
                    <form
                      onSubmit={[Function]}
                    >
                      <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="Search"
                        type="text"
                        value=""
                      />
                    </form>
                  </div>
                </div>
              </div>
            </BasicPatientSearch>
          </div>
        </div>
      </NavSearch>
    </Router>
  </StaticRouter>
</Provider>
`;

exports[`Component <NavSearch /> should renders correctly shallow 1`] = `
<div
  className="wrap-search wrap-header-search"
>
  <div
    className="header-search"
  >
    <div
      className="control-group left control-search-select dropdown"
    >
      <div>
        <SearchContent
          ADVANCED_SEARCH="advancedSearch"
          BASIC_SEARCH="basicSearch"
          SEARCH_CONTENT="searchContent"
          onSelect={[Function]}
          selected="advancedSearch"
        />
        <PTButton
          className="btn btn-dropdown-toggle btn-search-toggle"
          onClick={[Function]}
        >
          <i
            className="btn-icon fa fa-bars"
          />
        </PTButton>
      </div>
    </div>
    <Connect(AdvancedPatientSearch)
      className="advanced-search"
      onClose={[Function]}
    />
  </div>
</div>
`;
